### PR TITLE
SDK-3736 - React Native new architecture + iOS SDK 7.0.1 + Android SDK 7.0.1

### PR DESCRIPTION
**Pull request by kushCT**
##### Added
- Migrates the CleverTap bridge to a backwards-compatible [New Architecture Turbo Module](https://reactnative.dev/docs/next/the-new-architecture/pillars-turbomodules).
  - This is a non-breaking change to your existing imports of the CT SDK if you are using React Native `0.71`+.
  - The CleverTap SDK continues to be compatible with both the New Architecture and old React Native architecture.
  
  
https://github.com/react-native-community/RNNewArchitectureLibraries/tree/feat/back-turbomodule
https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-libraries-android.md

It also adds support for corev7.0.1 https://wizrocket.atlassian.net/browse/SDK-3969

### CODE REVIEW

File: src/NativeCleverTapModule.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  getDouble(
+    key: string,
+    callback: ((error: Object, result: boolean) => void) | null
+  ): void;
+  reset(): void;
+  suspendInAppNotifications(): void;
+  discardInAppNotifications(): void;
+  resumeInAppNotifications(): void;
+  fetchInApps(callback: ((error: Object, result: boolean) => void) | null): void;
+  clearInAppResources(expiredOnly: number): void;
```
Comment (Anush-Shand) [inline src/NativeCleverTapModule.ts:190]: `expiredOnly` could be a `boolean`

File: src/NativeCleverTapModule.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  promptForPushPermission(showFallbackSettings: number): void;
```
Comment (Anush-Shand) [inline src/NativeCleverTapModule.ts:191]: `showFallbackSettings` could be a` boolean`

**Comment by ankurbit1997** [comment]
@kushCT 
thanks for the amazing work on this 
can i try this on my project on new arch?

**Comment by darshanclevertap** [comment]
@ankurbit1997 - We are working on releasing a new version of the CleverTap RN SDK with the new architecture support. This is a PR for that enhancement. I would urge you to wait till we do the final release instead of using a dev branch for your app.

**Comment by ankurbit1997** [comment]
makes sense 
thanks @darshanclevertap 
can i ask the tentative date for this next launch.
so we can plan further

**Review by akashvercetti — APPROVED** [review]

File: android/src/oldarch/CleverTapModule.kt [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+        private const val CLEVERTAP_PRODUCT_CONFIG_DID_FETCH = "CleverTapProductConfigDidFetch"
+        private const val CLEVERTAP_PRODUCT_CONFIG_DID_ACTIVATE = "CleverTapProductConfigDidActivate"
+        private const val CLEVERTAP_PUSH_NOTIFICATION_CLICKED = "CleverTapPushNotificationClicked"
+        private const val CLEVERTAP_ON_PUSH_PERMISSION_RESPONSE = "CleverTapPushPermissionResponseReceived"
+        private const val CLEVERTAP_ON_VARIABLES_CHANGED = "CleverTapOnVariablesChanged"
+        private const val CLEVERTAP_ON_VALUE_CHANGED = "CleverTapOnValueChanged"
+    }
+
+    override fun getConstants(): Map<String, Any> {
+        val constants: MutableMap<String, Any> = HashMap()
```
[the remark this answers is not in the document]
Comment (Anush-Shand) [reply android/src/oldarch/CleverTapModule.kt]: Resolved in [771f9ae](https://github.com/CleverTap/clevertap-react-native/pull/396/commits/771f9ae1bd8b883a6727a61c873e830972dabc6d)

**Review by CTLalit — APPROVED** [review]

**Review by Anush-Shand — APPROVED** [review]

File: ios/CleverTapReact/CleverTapReactManager.mm [hunk 4]
```diff
@@ -17,7 +17,7 @@
 #import "CleverTapPushNotificationDelegate.h"
 #import "CleverTapInAppNotificationDelegate.h"
 #import "CleverTap+PushPermission.h"
-#import "CleverTapReactEventEmitter.h"
+//#import "CleverTapReactEventEmitter.h"
```
Comment (vasct) [inline ios/CleverTapReact/CleverTapReactManager.mm:20]: Probably not needed ?

File: src/index.js [hunk 5]
```diff
@@ -364,7 +366,7 @@ var CleverTap = {
     * Set the user's location as a latitude,longitude coordinate
     * @param {float} latitude
     * @param {float} longitude
-    */
+    // */
```
Comment (vasct) [inline src/index.js:369]: Too many comment markers

**Review by vasct — APPROVED** [review]

File: ios/CleverTapReact/CleverTapReactManager.mm (hunk 4 above)
Comment (akashvercetti) [reply ios/CleverTapReact/CleverTapReactManager.mm]: done

**Pull request merged**